### ISSUE TITLE
No logs if TestNG Test has timeout parameter

### ISSUE THREAD

**Issue by violant** [reportportal/logger-java-log4j#12]
I faced with no logs in RP if I specify timeout parameter. All events that are relayed to start/stop/methods reflected fine in RP, but no logging from TestNG test that I send through RP appender. 
Here is example of `@Test` annotation
`@Test(description = "test", dataProvider = "DataProvider", alwaysRun = true, **timeout = 60**)`

**Comment by dagansandler** [comment]
I think this relates to https://github.com/reportportal/client-java/issues/91

I suggested a (apparently naive) Pull Request at the time, but it was reverted after additional issues appeared:
https://github.com/reportportal/client-java/pull/53

Perhaps someone can pick this up and find a better solution

**Comment by HardNorth** [comment]
Yes, that's client issue due to ThreadLocal we use in our logging contexts. I believe we need to redesign it, but not in current (v5) version.
